File: benchmarks/pandas/bench_cat_rename_set_categories.py
"""Benchmark: cat_rename_set_categories — pandas Categorical rename_categories/set_categories on 100k-element Series"""
import json
import time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

cats = ["a", "b", "c", "d"]
s = pd.Categorical([cats[i % len(cats)] for i in range(ROWS)], categories=cats)

for _ in range(WARMUP):
    _ = s.rename_categories({"a": "alpha", "b": "beta"})
    _ = s.set_categories(["a", "b", "c", "d", "e"])

start = time.perf_counter()
for _ in range(ITERATIONS):
    _ = s.rename_categories({"a": "alpha", "b": "beta"})
    _ = s.set_categories(["a", "b", "c", "d", "e"])
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "cat_rename_set_categories", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_cat_rename_set_categories.ts
/**
 * Benchmark: cat_rename_set_categories — CategoricalAccessor renameCategories/setCategories on 100k-element Series
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const cats = ["a", "b", "c", "d"];
const s = new Series({ data: Array.from({ length: ROWS }, (_, i) => cats[i % cats.length]) });

for (let i = 0; i < WARMUP; i++) {
  s.cat.renameCategories({ a: "alpha", b: "beta" });
  s.cat.setCategories(["a", "b", "c", "d", "e"], false);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.cat.renameCategories({ a: "alpha", b: "beta" });
  s.cat.setCategories(["a", "b", "c", "d", "e"], false);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "cat_rename_set_categories",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
